common/src/main/java/com/wynntils/features/embellishments/ItemInteractionSoundsFeature.java

```diff
@@ -0,0 +1,228 @@
+    private static final SoundEvent GEM_ITEM_SOUND = newItemSound("gem-item");
+    private static final SoundEvent ITEM_BAG_SOUND = newItemSound("item-bag");
+    private static final SoundEvent ITEM_BOX_SOUND = newItemSound("item-box");
+    private static final SoundEvent METAL_ITEM_SOUND = newItemSound("metal-item");
+    private static final SoundEvent PAPER_ITEM_SOUND = newItemSound("paper-item");
+    private static final SoundEvent SADDLE_SOUND = newItemSound("saddle");
+    private static final SoundEvent STONE_ITEM_SOUND = newItemSound("stone-item");
+    private static final SoundEvent WOOD_ITEM_SOUND = newItemSound("wood-item");
+
+    private static final SoundEvent SPEAR_SOUND = newItemSound("weapon.spear");
+    private static final SoundEvent BOW_SOUND = newItemSound("weapon.bow");
+    private static final SoundEvent DAGGER_SOUND = newItemSound("weapon.dagger");
+    private static final SoundEvent WAND_SOUND = newItemSound("weapon.wand");
+    private static final SoundEvent RELIK_SOUND = newItemSound("weapon.relik");
+
+    private static final SoundEvent LEATHER_ARMOR_SOUND = newItemSound("armor.leather");
+    private static final SoundEvent CHAIN_ARMOR_SOUND = newItemSound("armor.chain");
+    private static final SoundEvent METAL_ARMOR_SOUND = newItemSound("armor.metal");
+
+    private static final SoundEvent RING_SOUND = newItemSound("accessory.ring");
+    private static final SoundEvent BRACELET_SOUND = newItemSound("accessory.bracelet");
+    private static final SoundEvent NECKLACE_SOUND = newItemSound("accessory.necklace");
+
+    private static final SoundEvent WOOD_TOOL_SOUND = newItemSound("tool.wood");
+    private static final SoundEvent STONE_TOOL_SOUND = newItemSound("tool.stone");
+    private static final SoundEvent METAL_TOOL_SOUND = newItemSound("tool.metal");
+
+    private static SoundEvent newItemSound(String key) {
+        return SoundEvent.createVariableRangeEvent(
+                ResourceLocation.fromNamespaceAndPath("wynntils", "inventory.item." + key));
+    }
+
+    @Persisted
+    private final Config<Float> soundVolume = new Config<>(1.0f);
+
+    @Persisted
+    private final Config<Boolean> mythicLayer = new Config<>(true);
+
+    @Persisted
+    private final Config<Boolean> shinyLayer = new Config<>(true);
+
+    @SubscribeEvent
+    public void onInventoryInteraction(InventoryInteractionEvent event) {
+        switch (event.getInteraction()) {
+            case InventoryInteraction.PickUp ixn -> playSoundEvent(PICK_UP_SOUND);
+            case InventoryInteraction.Place ixn -> playItemSound(ixn.placed());
+            case InventoryInteraction.Spread ixn -> playItemSound(ixn.stack());
+            case InventoryInteraction.Swap ixn -> playItemSound(ixn.placed());
+            case InventoryInteraction.Transfer ixn -> playItemSound(ixn.transferred());
+            default -> {}
+        }
+    }
+
+    public void playItemSound(ItemStack stack) {
```

Comment:
I find the overloaded name here a confusing and unsuitable. Combined with the switch above, where it is not easy to know what item type is returned from the events, that makes it impossible to see just by reading the code if e.g. InventoryInteraction.Spread calls this method or the other one. 

In general, I think name overloading is a mis-feature, and the only good usage is if the essentially same method but with a different set of arguments (or a different, but similar type, for one or more of the arguments.)